doc/cabal-package.rst

```diff
@@ -801,15 +803,17 @@ Library
 
     Build information for libraries.
 
-    Currently, there can only be one publicly exposed library in a
-    package, and its name is the same as package name set by global
-    :pkg-field:`name` field. In this case, the ``name`` argument to
-    the :pkg-section:`library` section must be omitted.
+    A package can have one default library, which is identified by omitting the
```

Comment:
Good, but there should be something in the sense of "name is the same as package name" from the previous text, because it's still true: the default library does have the same name, although it's not spelled out in the declaration.

Reply:
Technically the main library does not have a name. There is a misunderstanding due to how the dependency syntax evolved:

* With the legacy syntax, in `build-depends: pkg`, `pkg` is the package name. It gets interpreted as library name only if there is a library with that name in the same package, otherwise it's implied that the dependency is on the main/default/unnamed library.
* With the new syntax, in `build-depends: pkg:lib`, `pkg` is the package name and `lib` is the library name. To refer to the main/default/unnamed library, you omit the name, `:lib`. [It was decided](https://github.com/haskell/cabal/issues/4206#issuecomment-423436496) to also alias `pkg:pkg` to the main library because it fits better when using the extended `pkg:{lib1,lib2}` syntax and because [the name was already reserved](https://github.com/haskell/cabal/issues/4206#issuecomment-423641604).

Practically it doesn't matter much

Reply:
This is also explained in https://cabal.readthedocs.io/en/latest/cabal-package.html#pkg-field-build-depends, which is probably the appropriate place for it.